appcelerator/titanium_mobile
# [TIMOB-26218] Android: Resolved Veracode string equality warnings.

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-26218

**Summary:**
- Replaced string reference equality with value equality, as reported by Veracode.

**Test:**
1. Build and run the below code on an Android device with a camera.
2. Tap the "Take Picture" button.
3. If a dialog requests permission, allow it.
4. Take a picture with the camera.
5. When returning from to the app from the camera, verify that app displays the camera shot.
6. Verify that the log outputs: `@@@ Camera image size: <width>x<height>`

```javascript
var window = Ti.UI.createWindow();
var imageView = Ti.UI.createImageView();
window.add(imageView);
var button = Ti.UI.createButton({
	title: "Take Picture",
	bottom: "10dp",
});
button.addEventListener("click", function(e) {
	if (!Ti.Media.hasCameraPermissions()) {
		Ti.Media.requestCameraPermissions(function(e) {
			if (e.success) {
				button.fireEvent("click", {});
			}
		});
	} else if (!Ti.Filesystem.hasStoragePermissions()) {
		Ti.Filesystem.requestStoragePermissions(function(e) {
			if (e.success) {
				button.fireEvent("click", {});
			}
		});
	} else {
		Ti.Media.showCamera({
			mediaTypes: [Ti.Media.MEDIA_TYPE_PHOTO],
			success: function(e) {
				Ti.API.info("@@@ Camera image size: " + e.media.width + "x" + e.media.height);
				imageView.image = e.media;
			},
			cancel: function(e) {
				Ti.API.info("@@@ Camera canceled");
			},
			error: function(e) {
				Ti.API.info("@@@ Camera error: " + e.error);
				imageView.image = null;
			}
		});
	}
});
window.add(button);
window.open();
```
